spec/services/login/user_verifier_spec.rb

```diff
@@ -280,7 +280,36 @@
           end
 
           context 'and user_account matching icn already exists' do
-            let!(:existing_user_account) { UserAccount.create!(icn:) }
+            let!(:existing_user_account) { UserAccount.create(icn:) }
+
+            context 'and a linked user verification with the same CSP type exists' do
+              let!(:linked_user_verification) do
+                case login_value
```

Comment:
I generally don't think we should have conditional in `let` statements if possible. Instead, put the proper `linked_user_verification` in the `context` that is calling the `shared_example`.

This means you can just add the appropriate `linked_user_verification` directly under `when user credential is mhv`, `when user credential is dslogon`, etc